File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPVerifyAgentInstall.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class CSPVerifyAgentInstall
    {
        public void run(Dictionary<string, object> Params)
        {
            var service = new CSPAgentService();
            
            if (!service.IsInstalled()) {
                throw new Exception("The agent service has not been installed!");
            }
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPVerifyAgentInstall.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.util.*;

import com.infor.cloudsuite.portal.singletenant.agents.services.*;

class CSPVerifyAgentInstall
{
    public void run(HashMap<String, Object> params) throws Exception
    {
        CSPAgentServiceController agentServiceController = null;
        
        if (System.getProperty("os.name").contains("Windows")) {
            agentServiceController = new CSPWindowsAgentServiceController();
        }
        else if (System.getProperty("os.name").contains("Linux")) {
            agentServiceController = new CSPLinuxAgentServiceController();
        }
        
        if (!agentServiceController.isInstalled()) { throw new Exception("The agent service has not been installed!"); }
    }
}